src/nrnoc/init.cpp

```diff
@@ -772,30 +768,24 @@ namespace {
  *
  * This logic used to live inside hoc_register_dparam_semantics.
  */
+
+// name to int map for the negative types
+// xx_ion and #xx_ion will get values of type and type+1000 respectively
+static std::unordered_map<std::string, int> name_to_negint = {{"area", -1},
+                                                              {"iontype", -2},
+                                                              {"cvodeieq", -3},
+                                                              {"netsend", -4},
+                                                              {"pointer", -5},
+                                                              {"pntproc", -6},
+                                                              {"bbcorepointer", -7},
+                                                              {"watch", -8},
+                                                              {"diam", -9},
+                                                              {"fornetcon", -10},
+                                                              {"random", -11}};
+
 int dparam_semantics_to_int(std::string_view name) {
-    // only interested in area, iontype, cvode_ieq, netsend, pointer, pntproc, bbcorepointer, watch,
-    // diam, fornetcon, xx_ion and #xx_ion which will get a semantics value of -1, -2, -3, -4, -5,
-    // -6, -7, -8, -9, -10 type, and type+1000 respectively
-    if (name == "area") {
-        return -1;
-    } else if (name == "iontype") {
-        return -2;
-    } else if (name == "cvodeieq") {
-        return -3;
-    } else if (name == "netsend") {
-        return -4;
-    } else if (name == "pointer") {
-        return -5;
-    } else if (name == "pntproc") {
-        return -6;
-    } else if (name == "bbcorepointer") {
-        return -7;
-    } else if (name == "watch") {
-        return -8;
-    } else if (name == "diam") {
-        return -9;
-    } else if (name == "fornetcon") {
-        return -10;
+    if (auto got = name_to_negint.find(std::string{name}); got != name_to_negint.end()) {
```

Comment:
yeah - with an enum you'd avoid doing these string operations.

Reply:
I like the enum style.  It just leaves out the two situations where the name or ```#name``` is not
in the enum list. That is taken care of by  this function ```int dparam_semantics_to_int(std::string_view name)``` but, then, what is the arg and return
value supposed to be.
The arg can't be an enum element since that leaves out all the ```ionname``` and ```#ionname``` possibilities. And the return value can't be an emum for the same reason. I.e currently we have an integer semantic code that ranges from -11 to -1, all of the positive ion mechtypes, and 1000+(all of the positive ion mechtypes).

Reply:
ok, I see - yes, I missed that a lot more gets encoded in there currently.